### ISSUE TITLE
Deserialization Issue (PaperSpigot 1.12.2)

### ISSUE THREAD

**Issue by DavidVioMC** [DutchJelly/CraftEnhance#6]
```
[03:07:34] [Server thread/ERROR]: Error occurred while enabling CraftEnhance v2.3.3 (Is it up to date?)
java.lang.ClassCastException: org.bukkit.configuration.MemorySection cannot be cast to com.dutchjelly.craftenhance.model.CraftRecipe
	at com.dutchjelly.craftenhance.files.FileManager.cacheRecipes(FileManager.java:76) ~[?:?]
	at com.dutchjelly.craftenhance.CraftEnhance.setupFileManager(CraftEnhance.java:130) ~[?:?]
	at com.dutchjelly.craftenhance.CraftEnhance.onEnable(CraftEnhance.java:56) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[patched_1.12.2.jar:git-Paper-1618]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:316) ~[patched_1.12.2.jar:git-Paper-1618]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:405) ~[patched_1.12.2.jar:git-Paper-1618]
	at org.bukkit.craftbukkit.v1_12_R1.CraftServer.enablePlugin(CraftServer.java:395) ~[patched_1.12.2.jar:git-Paper-1618]
	at org.bukkit.craftbukkit.v1_12_R1.CraftServer.enablePlugins(CraftServer.java:344) ~[patched_1.12.2.jar:git-Paper-1618]
	at net.minecraft.server.v1_12_R1.MinecraftServer.t(MinecraftServer.java:442) ~[patched_1.12.2.jar:git-Paper-1618]
	at net.minecraft.server.v1_12_R1.MinecraftServer.l(MinecraftServer.java:403) ~[patched_1.12.2.jar:git-Paper-1618]
	at net.minecraft.server.v1_12_R1.MinecraftServer.a(MinecraftServer.java:341) ~[patched_1.12.2.jar:git-Paper-1618]
	at net.minecraft.server.v1_12_R1.DedicatedServer.init(DedicatedServer.java:289) ~[patched_1.12.2.jar:git-Paper-1618]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:616) ~[patched_1.12.2.jar:git-Paper-1618]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_242]
[03:07:34] [Server thread/INFO]: [CraftEnhance] Disabling CraftEnhance v2.3.3
[03:07:34] [Server thread/ERROR]: Error occurred while disabling CraftEnhance v2.3.3 (Is it up to date?)
java.lang.NullPointerException: null
	at com.dutchjelly.craftenhance.CraftEnhance.onDisable(CraftEnhance.java:74) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:266) ~[patched_1.12.2.jar:git-Paper-1618]
	at org.bukkit.plugin.java.JavaPluginLoader.disablePlugin(JavaPluginLoader.java:350) ~[patched_1.12.2.jar:git-Paper-1618]
	at org.bukkit.plugin.SimplePluginManager.disablePlugin(SimplePluginManager.java:437) ~[patched_1.12.2.jar:git-Paper-1618]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:320) ~[patched_1.12.2.jar:git-Paper-1618]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:405) ~[patched_1.12.2.jar:git-Paper-1618]
	at org.bukkit.craftbukkit.v1_12_R1.CraftServer.enablePlugin(CraftServer.java:395) ~[patched_1.12.2.jar:git-Paper-1618]
	at org.bukkit.craftbukkit.v1_12_R1.CraftServer.enablePlugins(CraftServer.java:344) ~[patched_1.12.2.jar:git-Paper-1618]
	at net.minecraft.server.v1_12_R1.MinecraftServer.t(MinecraftServer.java:442) ~[patched_1.12.2.jar:git-Paper-1618]
	at net.minecraft.server.v1_12_R1.MinecraftServer.l(MinecraftServer.java:403) ~[patched_1.12.2.jar:git-Paper-1618]
	at net.minecraft.server.v1_12_R1.MinecraftServer.a(MinecraftServer.java:341) ~[patched_1.12.2.jar:git-Paper-1618]
	at net.minecraft.server.v1_12_R1.DedicatedServer.init(DedicatedServer.java:289) ~[patched_1.12.2.jar:git-Paper-1618]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:616) ~[patched_1.12.2.jar:git-Paper-1618]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_242]
```

**Comment by DutchJelly** [comment]
When submitting an issue you should clearly show how to reproduce the issue.

These are some questions that could help clarifying the issue:
- what does your recipes.yml look like?
- what's your server version?
- when did the issue first occur?

**Comment by DutchJelly** [comment]
I think this is fixed in `1.4.0`, if not please open a new issue.
